File: notebook_version_cpp/person.cpp
#include <iostream>
#include <string>
#include <fstream>
#include "person.h"
   
    Person::Person(std::string name, std::string number, std::string email, int age)
    {
        Name = name;
        Number = number;
        Email = email;
        Age = age;
    }
    Person::Person(std::string name, std::string number, std::string email) //: Person::Person(name, number, email, 0)
    {
        Name = name;
        Number = number;
        Email = email;
        Age = 0;
    }
    Person::Person(std::string name, std::string number) //: Person::Person(name, number, "no email", 0)
    {
        Name = name;
        Number = number;
        Email = "no email";
        Age = 0;
    }
    Person::Person(std::string name) //: Person::Person(name, "no number", "no email", 0)
    {
        throw "Необходимо ввести номер телефона или электронную почту для создания контакта";
    }
    Person::Person() //: Person::Person("no name", "no number", "no email", 0)
    {
        throw "Необходимо ввести номер телефона или электронную почту для создания контакта";
    }

    void Person::GetInfo()
    {
        std::cout<<"Информация о контакте:"<<std::endl;
        std::cout<<"Имя: "<<Name<<std::endl;
        std::cout<<"Возраст: "<<Age<<std::endl;
        std::cout<<"Номер: "<<Number<<std::endl;
        std::cout<<"Электронная почта: "<<Email<<std::endl;
    }

File: notebook_version_cs/Person.cs
using System;

namespace Notebook
{
    sealed public class Person
    {
        public int Id { get; set; }
        public int age { get; set; }
        public string name { get; set; }
        public string number { get; set; }
        public string email { get; set; }

        public Person():this("not number")
        {
            throw new Exception("Нельзя создать контакт\n Необходим номер");
        }
        public Person(string number):this(number,"no name")
        {

        }
        public Person(string number, string name):this(number, name, 0)
        {

        }
        public Person(string number, string name, int age) : this(number, name, age, "example@mail.com")
        {

        }
        public Person(string number, string name, int age, string email)
        {
            this.age = age;
            this.name = name;
            this.number = number;
            this.email = email;
        }

        public void GetInfo()
        {
            Console.WriteLine(age);
            Console.WriteLine(name);
            Console.WriteLine(number);
            Console.WriteLine(email);
        }
    }
}

